content/blog/trustless-lightning-payments-for-content.md

```diff
@@ -0,0 +1,357 @@
+
+Unfortunately, this approach requires the consumer to trust the honesty of the
+content seller. A malicious seller could create an invoice using an invalid
+decryption key as the preimage. The buyer would make the Lightning payment, and
+only realize after the payment that they got an invalid decryption key. The
+buyer has no way to verify the validity of the invoice until it is too late,
+and they have already lost their funds.
+
+This model of monetization can only work if the buyer is able to trust the seller.
+This might be true if the seller is a large, reputable institution, but this
+model breaks down if we try to extend it to a peer-to-peer decentralized
+network, where all connections are assumed to be trustless.
+
+#### Making payments trustless with PTLCs
+
+Fortunately, there is a way to make the payment trustless for both the buyer
+and the seller, but it requires that Lightning use PTLC payment channels.
+
+With PTLCs, the seller specifies a 32-byte scalar value (`s`) as the "preimage"
+at the creation-time of the invoice. When the buyer decodes the invoice payment
+string, they will obtain the payment point (`p = s*G`). The buyer will only
+obtain the "preimage" `s` after the payment is settled.
+
+So now that we know that PTLC invoices use elliptic curve points, we can
+take advantage of the distributive property of points on elliptic curves:
+
+```
+(s1 + s2)*G = s1*G + s2*G
+```
+
+where `s1` and `s2` are scalars, and `G` is an elliptic curve.
+
+The basic idea for selling content is as follows:
+
+* Alice has a piece of content she wants to be sellable.
+* Alice generates a scalar value `s1` to use as a symmetric encryption/decryption key, and encrypts the content.
+* Alice calculates the point `p1` on an elliptic curve `G` by calculating `p1 = s1*G`.
+* Alice makes `p1` publicly available for anyone interested in buying the content.
+* Alice shares the encrypted content with a relay named Carol.
+* Bob downloads the encrypted content from Carol, and requests an invoice to unlock the content.
+* Carol generates a new scalar value `s2`, and creates a *PTLC* Lightning invoice with `s1 + s2` as the preimage.
+* Carol sends Bob `s2` and the Lightning invoice as a payment request string.
+* Bob decodes the payment request string to get the payment point of the invoice, call it `p3`.
+* Bob calculates `s2*G`. If it is equal to `p3 - p1`, then Bob knows that the invoice is valid.
+* Bob pays the Lightning invoice, and gets the value of `s1 + s2` as the preimage.
+* Bob calculates `s1 = (s1 + s2) - s2` to get the decryption key, and decrypts the content.
+
+If Carol tries to give Bob an invalid invoice, Bob will know before he pays the invoice.
+
+The above example assumes that Bob is able to trust that the value of `p1` is valid.
+
+{{< center-figure src="/post-data/trustless-lightning-payments/ptlc-example-2.png" caption="" height=600 width=600 >}}
```

Comment:
Crop transparent padding and (optionally) add a caption.

![image](https://user-images.githubusercontent.com/19157360/149501866-ead83b31-7213-4dec-a5b3-53d565cbe9a4.png)

Reply:
Fixed.